recipes-bsp/u-boot/files/0006-dts-iot2050-Support-new-IOT2050-SM-variant.patch

```diff
@@ -0,0 +1,634 @@
++&mcu_spi0 {
++	pinctrl-names = "default";
++	pinctrl-0 = <&mcu_spi0_pins_default>;
++};
++
++&main_i2c3 {
++	pinctrl-names = "default";
++	pinctrl-0 = <&main_i2c3_pins_default>;
++	clock-frequency = <400000>;
++
++	accelerometer: lsm6dso@6a {
++		compatible = "st,lsm6dso";
++		reg = <0x6a>;
++	};
++
++	lightsensor: pm16d17@44 {
++		compatible = "everlight,pm16d17";
++		reg = <0x44>;
++
++		ps-gain = <1>;
++		ps-itime = "0.4";
++		ps-wtime = "25";
++		ps-ir-led-pulse-count = <1>;
++	};
++
++	/delete-node/ edp-bridge@f;
++};
++
++/delete-node/ &pcal9535_1;
++/delete-node/ &pcal9535_2;
++/delete-node/ &pcal9535_3;
++
++&dss {
++	status = "disabled";
++};
++
++&dss_ports {
++	/delete-node/ port@1;
++};
++
++&ecap0 {
++	status = "disabled";
++};
++
++&mcu_uart0 {
++	status = "disabled";
++};
++
++&tscadc1 {
++	status = "disabled";
++};
++
++&serdes0 {
++	assigned-clocks = <&k3_clks 153 4>, <&serdes0 AM654_SERDES_CMU_REFCLK>;
++	assigned-clock-parents = <&k3_clks 153 8>, <&k3_clks 153 4>;
++};
++
++&serdes1 {
++	status = "disabled";
++};
++
++&pcie0_rc {
++	pinctrl-names = "default";
++	pinctrl-0 = <&minipcie_pins_default>;
++
++	num-lanes = <1>;
++	phys = <&serdes0 PHY_TYPE_PCIE 1>;
++	phy-names = "pcie-phy0";
++	reset-gpios = <&wkup_gpio0 27 GPIO_ACTIVE_HIGH>;
++	status = "okay";
++};
++
++&pcie0_ep {
++	status = "disabled";
++};
++
++&pcie1_rc {
++	status = "disabled";
++};
++
++&pcie1_ep {
++	status = "disabled";
++};
++
++&dwc3_0 {
++	assigned-clock-parents = <&k3_clks 151 4>,  /* set REF_CLK to 20MHz i.e. PER0_PLL/48 */
++				 <&k3_clks 151 9>;  /* set PIPE3_TXB_CLK to CLK_12M_RC/256 (for HS only) */
++	/delete-property/ phys;
++	/delete-property/ phy-names;
++};
++
++&usb0 {
++	pinctrl-names = "default";
++	pinctrl-0 = <
++		&usb0_pins_default
++	>;
++
++	maximum-speed = "high-speed";
++	/delete-property/ snps,dis-u1-entry-quirk;
++	/delete-property/ snps,dis-u2-entry-quirk;
++};
+diff --git a/board/siemens/iot2050/board.c b/board/siemens/iot2050/board.c
+index 0b0686e2628b..d0d8b8d75700 100644
+--- a/board/siemens/iot2050/board.c
++++ b/board/siemens/iot2050/board.c
+@@ -38,6 +38,8 @@ struct iot2050_info {
+ 	u8 mac_addr_cnt;
+ 	u8 mac_addr[8][ARP_HLEN];
+ 	char seboot_version[40 + 1];
++	u8 padding[3];
++	u32 ddr_size_mb;
+ } __packed;
+ 
+ /*
+@@ -172,6 +174,14 @@ static bool board_is_m2(void)
+ 		strcmp((char *)info->name, "IOT2050-ADVANCED-M2") == 0;
+ }
+ 
++static bool board_is_sm(void)
++{
++	struct iot2050_info *info = IOT2050_INFO_DATA;
++
++	return info->magic == IOT2050_INFO_MAGIC &&
++		strcmp((char *)info->name, "IOT2050-ADVANCED-SM") == 0;
++}
++
+ static void remove_mmc1_target(void)
+ {
+ 	char *boot_targets = strdup(env_get("boot_targets"));
+@@ -185,6 +195,15 @@ static void remove_mmc1_target(void)
+ 	free(boot_targets);
+ }
+ 
++static void enable_pcie_connector_power(void)
++{
++	if (board_is_sm())
++		set_pinvalue("gpio@601000_22", "P3V3_PCIE_CON_EN", 1);
++	else
++		set_pinvalue("gpio@601000_17", "P3V3_PCIE_CON_EN", 1);
```

Comment:
The label name changes here as well for existing M.2 - harmless?

Reply:
And this code is now run for all boards, not just M.2 as so far.

Reply:
The plan is to control the 3.3V power pin of M.2 or miniPCIE interface, for all variants, to workaround some power timing issues observed on some cards. (Hardware change will be introduced one by one.)

The label name change is harmless. Since there is word 'M2' in the old name, which does not fit the fact of other variants. In general, either miniPCIE or M.2, is kind of PCIE connection, so maybe 'PCIE_CON' is a more suitable name. - There is indeed other kind of interfaces in these two connections, such as USB2/3 ... Any suggestion for a better naming?

Reply:
Separated.